### PR TITLE
Enhanced the script to run for PowerNV systmes as well

### PR DESCRIPTION
**Pull request by vrbagalkote**
Earlier script used to fail for PowerNV as the slots listed in
PowerNV machine is different than in PowerVM, and its handeled now
properly.

Signed-off-by: root <root@ltc-tuleta4u-l.aus.stglabs.ibm.com>

### CODE REVIEW

**Comment by vrbagalkote** [comment]
[pci_hotplug.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1138797/pci_hotplug.txt)

Job.log file

File: io/pci/pci_hotplug.py [hunk 1]
```diff
@@ -42,28 +42,35 @@ def setUp(self):
...
+            self.slot = genio.read_file("/proc/device-tree/%s/ibm,loc-code"
+                                        % devspec)
+            self.slot = re.match(r'((\w+)[\.])+(\w+)-P(\d+)-C(\d+)|Slot(\d+)',
+                                 self.slot).group()
+        else:
+            self.slot = pci.get_pci_prop(self.device, "PhySlot")
+            print self.slot
         if not os.path.isdir('/sys/bus/pci/slots/%s' % self.slot):
-            self.cancel("%s Slot not available" % self.slot)
+            self.skip("%s Slot not available" % self.slot)
```
Comment (narasimhan-v) [inline io/pci/pci_hotplug.py]: `self.cancel()` please

File: io/pci/pci_hotplug.py [hunk 2]
```diff
@@ -42,28 +42,35 @@ def setUp(self):
...
         self.device = self.params.get('pci_device', default=' ')
         if not os.path.isdir('/sys/bus/pci/devices/%s' % self.device):
-            self.cancel("PCI device given does not exist")
-        devspec = genio.read_file("/sys/bus/pci/devices/%s/devspec"
-                                  % self.device)
-        self.slot = genio.read_file("/proc/device-tree/%s/ibm,loc-code"
-                                    % devspec)
-        self.slot = re.match(r'((\w+)[\.])+(\w+)-P(\d+)-C(\d+)|Slot(\d+)',
-                             self.slot).group()
+            self.skip("PCI device given does not exist")
```
Comment (narasimhan-v) [inline io/pci/pci_hotplug.py]: `self.cancel()`

File: io/pci/pci_hotplug.py [hunk 3]
```diff
@@ -42,28 +42,35 @@ def setUp(self):
         """
         cmd = "uname -p"
         if 'ppc' not in process.system_output(cmd, ignore_status=True):
-            self.cancel("Processor is not ppc64")
-        if cpu._list_matches(cpu._get_cpu_info(), 'pSeries'):
+            self.skip("Processor is not ppc64")
+        cmd = "cat /proc/cpuinfo"
```
Comment (narasimhan-v) [inline io/pci/pci_hotplug.py]: Was the previous implementation of using `cpu._get_cpu_info()` not working ?

Comment (vrbagalkote) [inline io/pci/pci_hotplug.py]: yes, it was not working

Comment (narasimhan-v) [inline io/pci/pci_hotplug.py]: Ah, ok

File: io/pci/pci_hotplug.py [hunk 4]
```diff
@@ -42,28 +42,35 @@ def setUp(self):
...
+        if 'pSeries' in process.system_output(cmd, ignore_status=True):
```
Comment (narasimhan-v) [inline io/pci/pci_hotplug.py]: @abdhaleegit That was his exact intention here. To differentiate PowerVM and PowerNV

**Comment by vrbagalkote** [comment]
thanks @PraveenPenguin @narasimhan-v @abdhaleegit for you comments and thoughts. I have addressed the comments, and resent the patch.

**Comment by vrbagalkote** [comment]
sorry, my bad...

**Comment by vrbagalkote** [comment]
[pci_hotplug.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1141354/pci_hotplug.txt)

Job.log file with latest code

**Comment by PraveenPenguin** [comment]
@narasimhan-v @abdhaleegit thanks for review

**Pull request merged**